File: clients/node/sdk/models/context.ts
import {createStatusFromDiscriminatorValue} from './createStatusFromDiscriminatorValue';
import {Status} from './index';
import {AdditionalDataHolder, Parsable, ParseNode, SerializationWriter} from '@microsoft/kiota-abstractions';

export class Context implements AdditionalDataHolder, Parsable {
    /** Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well. */
    private _additionalData: Record<string, unknown>;
    /** The ancestors property */
    private _ancestors?: Status[] | undefined;
    /** The descendants property */
    private _descendants?: Status[] | undefined;
    /**
     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @returns a Record<string, unknown>
     */
    public get additionalData() {
        return this._additionalData;
    };
    /**
     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    public set additionalData(value: Record<string, unknown>) {
        this._additionalData = value;
    };
    /**
     * Gets the ancestors property value. The ancestors property
     * @returns a Status
     */
    public get ancestors() {
        return this._ancestors;
    };
    /**
     * Sets the ancestors property value. The ancestors property
     * @param value Value to set for the ancestors property.
     */
    public set ancestors(value: Status[] | undefined) {
        this._ancestors = value;
    };
    /**
     * Instantiates a new Context and sets the default values.
     */
    public constructor() {
        this._additionalData = {};
    };
    /**
     * Gets the descendants property value. The descendants property
     * @returns a Status
     */
    public get descendants() {
        return this._descendants;
    };
    /**
     * Sets the descendants property value. The descendants property
     * @param value Value to set for the descendants property.
     */
    public set descendants(value: Status[] | undefined) {
        this._descendants = value;
    };
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    public getFieldDeserializers() : Record<string, (node: ParseNode) => void> {
        return {
            "ancestors": n => { this.ancestors = n.getCollectionOfObjectValues<Status>(createStatusFromDiscriminatorValue); },
            "descendants": n => { this.descendants = n.getCollectionOfObjectValues<Status>(createStatusFromDiscriminatorValue); },
        };
    };
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    public serialize(writer: SerializationWriter) : void {
        if(!writer) throw new Error("writer cannot be undefined");
        writer.writeCollectionOfObjectValues<Status>("ancestors", this.ancestors);
        writer.writeCollectionOfObjectValues<Status>("descendants", this.descendants);
        writer.writeAdditionalData(this.additionalData);
    };
}

File: clients/python/sdk/models/context.py
from __future__ import annotations
from kiota_abstractions.serialization import AdditionalDataHolder, Parsable, ParseNode, SerializationWriter
from typing import Any, Callable, Dict, List, Optional, Union

from . import status

class Context(AdditionalDataHolder, Parsable):
    @property
    def additional_data(self,) -> Dict[str, Any]:
        """
        Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        Returns: Dict[str, Any]
        """
        return self._additional_data
    
    @additional_data.setter
    def additional_data(self,value: Dict[str, Any]) -> None:
        """
        Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        Args:
            value: Value to set for the AdditionalData property.
        """
        self._additional_data = value
    
    @property
    def ancestors(self,) -> Optional[List[status.Status]]:
        """
        Gets the ancestors property value. The ancestors property
        Returns: Optional[List[status.Status]]
        """
        return self._ancestors
    
    @ancestors.setter
    def ancestors(self,value: Optional[List[status.Status]] = None) -> None:
        """
        Sets the ancestors property value. The ancestors property
        Args:
            value: Value to set for the ancestors property.
        """
        self._ancestors = value
    
    def __init__(self,) -> None:
        """
        Instantiates a new Context and sets the default values.
        """
        # Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        self._additional_data: Dict[str, Any] = {}

        # The ancestors property
        self._ancestors: Optional[List[status.Status]] = None
        # The descendants property
        self._descendants: Optional[List[status.Status]] = None
    
    @staticmethod
    def create_from_discriminator_value(parse_node: Optional[ParseNode] = None) -> Context:
        """
        Creates a new instance of the appropriate class based on discriminator value
        Args:
            parseNode: The parse node to use to read the discriminator value and create the object
        Returns: Context
        """
        if parse_node is None:
            raise Exception("parse_node cannot be undefined")
        return Context()
    
    @property
    def descendants(self,) -> Optional[List[status.Status]]:
        """
        Gets the descendants property value. The descendants property
        Returns: Optional[List[status.Status]]
        """
        return self._descendants
    
    @descendants.setter
    def descendants(self,value: Optional[List[status.Status]] = None) -> None:
        """
        Sets the descendants property value. The descendants property
        Args:
            value: Value to set for the descendants property.
        """
        self._descendants = value
    
    def get_field_deserializers(self,) -> Dict[str, Callable[[ParseNode], None]]:
        """
        The deserialization information for the current model
        Returns: Dict[str, Callable[[ParseNode], None]]
        """
        fields = {
            "ancestors": lambda n : setattr(self, 'ancestors', n.get_collection_of_object_values(status.Status)),
            "descendants": lambda n : setattr(self, 'descendants', n.get_collection_of_object_values(status.Status)),
        }
        return fields
    
    def serialize(self,writer: SerializationWriter) -> None:
        """
        Serializes information the current object
        Args:
            writer: Serialization writer to use to serialize this model
        """
        if writer is None:
            raise Exception("writer cannot be undefined")
        writer.write_collection_of_object_values("ancestors", self.ancestors)
        writer.write_collection_of_object_values("descendants", self.descendants)
        writer.write_additional_data_value(self.additional_data)
    

